Classes/Command/FluidPrecompileCommandController.php

```diff
@@ -0,0 +1,220 @@
+<?php
+namespace NamelessCoder\CmsFluidPrecompilerModule\Command;
+
+use NamelessCoder\CmsFluidPrecompilerModule\Service\FluidPrecompilerService;
+use Symfony\Component\Console\Helper\TableSeparator;
+use TYPO3\CMS\Core\Utility\GeneralUtility;
+use TYPO3\CMS\Extbase\Mvc\Controller\CommandController;
+use TYPO3\CMS\Extbase\Object\ObjectManager;
+use TYPO3Fluid\Fluid\Core\Cache\FluidCacheWarmupResult;
+use TYPO3Fluid\Fluid\Core\Rendering\RenderingContextInterface;
+
+/**
+ * Class FluidPrecompileCommandController
+ */
+class FluidPrecompileCommandController extends CommandController
+{
+    /**
+     * Precompile fluid templates
+     *
+     * (Pre-)compiles all fluid templates to static files of the given or all extension(s).
+     * 
+     * @param string $extension extension key of which the fluid template should be compiled, leave empty to compile all
+     * @param bool $fail fail with a non-zero return code if any template could not be compiled
+     * @param bool $onlyFailed show only failed fluid templates from compiling
+     * @param int $limit limit of template files of each extension shown
+     * @throws \Exception
+     */
+    public function compileCommand($extension = null, $fail = true, $onlyFailed = true, $limit = null)
+    {
+        $lastError = null;
+
+        try {
+            $result = $this->getPrecompilerService()->warmup($extension);
+        } catch (\Exception $e) {
+            $result = [];
```

Comment:
This looks like a misunderstanding about what `$fail` actually does.

It does not suppress errors. It makes the warmup fail if any one template could not compile - which when it happens, does not raise an exception. So no try-catch here, but instead, a `throw` down below if there is one or more than failed templates.

Reply:
Sorry, I can't quite follow. Exceptions will be thrown at the end of the function if `--fail` was provided.

I understand the `--fail` flag as following:
If there's any kind of an error (e.g. compiling, TYPO3 or "uncompilable templates") **and** the `--fail` flag was provided the program should exit with a non-zero return code. In this case a thrown exception would cause this.
But when the `--fail` flag isn't provided it shouldn't, because the program was told not to fail.

Or would you like to see the application "failing" if an internal error (i.e. the compiling failed) occurs while staying silent for all contextual error (i.e. "uncompilable templates") when no `--fail` flag was provided.

Reply:
I've reworked the `--fail` logic:
Exceptions thrown while compiling/warmup won't be catched anymore.
Only if `--faill` was provided **and** there were uncompilable templates, a custom exception will be thrown which causes a non-zero status code.

See https://github.com/NamelessCoder/typo3-cms-fluid-precompiler-module/pull/6/commits/7215c5d13366cdd2d56357e7cb58a0732ec595a1